src/main/scala/io/appthreat/atom/Atom.scala

```diff
@@ -425,130 +448,128 @@ object Atom:
+                                  ".*src/test.*" r
                                 )
-                          )
-                          .map { ag =>
-                              new PythonImportsPass(ag).createAndApply()
-                              new PyImportResolverPass(ag).createAndApply()
-                              new DynamicTypeHintFullNamePass(ag).createAndApply()
-                              new PythonInheritanceNamePass(ag).createAndApply()
-                              new PythonTypeRecoveryPass(
-                                ag,
-                                XTypeRecoveryConfig(enabledDummyTypes = false)
                               )
-                                  .createAndApply()
-                              new PythonTypeHintCallLinker(ag).createAndApply()
-                              new AstLinkerPass(ag).createAndApply()
-                              ag
-                          }
-                  case Languages.PHP =>
-                      new Php2Atom().createCpgWithOverlays(
-                        PhpConfig()
-                            .withDisableDummyTypes(true)
-                            .withInputPath(config.inputPath.pathAsString)
-                            .withOutputPath(outputAtomFile)
-                            .withDefaultIgnoredFilesRegex(List("\\..*".r))
-                            .withIgnoredFilesRegex(".*(samples|examples|docs|tests).*")
-                      ).map { ag =>
-                          new PhpSetKnownTypesPass(ag).createAndApply()
-                          ag
-                      }
-                  case _ => Failure(
-                        new RuntimeException(
-                          s"No language frontend supported for language '$language'"
+                              .withOutputPath(outputAtomFile)
+                        )
+                case Languages.JSSRC | Languages.JAVASCRIPT | "JS" | "TS" | "TYPESCRIPT" =>
+                    new JsSrc2Cpg()
+                        .createCpgWithOverlays(
+                          JSConfig()
+                              .withDisableDummyTypes(true)
+                              .withTypePropagationIterations(TYPE_PROPAGATION_ITERATIONS)
+                              .withInputPath(config.inputPath.pathAsString)
+                              .withOutputPath(outputAtomFile)
+                        )
+                        .map { ag =>
+                            new JavaScriptInheritanceNamePass(ag).createAndApply()
+                            new ConstClosurePass(ag).createAndApply()
+                            new ImportResolverPass(ag).createAndApply()
+                            new JavaScriptTypeRecoveryPass(ag).createAndApply()
+                            new TypeHintPass(ag).createAndApply()
+                            ag
+                        }
+                case Languages.PYTHONSRC | Languages.PYTHON | "PY" =>
```

Comment:
@prabhu I'm just curious: What's the exact change here? This code seems precisely the same as the previous code, right?